python/lsst/qa/explorer/functors.py

```diff
@@ -222,85 +332,279 @@ def columns(self):
     def _func(self, df):
         return df[self.col]
 
+
+class Index(Functor):
+    """Return the value of the index for each object
+    """
+
+    columns = ['coord_ra']  # just a dummy; something has to be here
+    _default_dataset = 'ref'
+
+    def _func(self, df):
+        return pd.Series(df.index, index=df.index)
+
+
 class IDColumn(Column):
     col = 'id'
     _allow_difference = False
 
+
 class FootprintNPix(Column):
     col = 'base_Footprint_nPix'
 
+
 class CoordColumn(Column):
-    """Base class for coordinate column
+    """Base class for coordinate column, in degress
     """
     _allow_difference = False
     _radians = True
 
     def __init__(self, col, calculate=False, **kwargs):
         self.calculate = calculate
-        super(CoordColumn, self).__init__(col, **kwargs)
+        super().__init__(col, **kwargs)
 
     def _func(self, df):
         res = df[self.col]
         if self._radians:
             res *= 180 / np.pi
         return res
 
+
 class RAColumn(CoordColumn):
+    """Right Ascension, in degrees
+    """
     name = 'RA'
+
     def __init__(self, **kwargs):
-        super(RAColumn, self).__init__('coord_ra', **kwargs)
+        super().__init__('coord_ra', **kwargs)
 
     def __call__(self, catalog, **kwargs):
-        return super(RAColumn, self).__call__(catalog, **kwargs)
+        return super().__call__(catalog, **kwargs)
+
 
 class DecColumn(CoordColumn):
+    """Declination, in degrees
+    """
     name = 'Dec'
+
     def __init__(self, **kwargs):
-        super(DecColumn, self).__init__('coord_dec', **kwargs)
+        super().__init__('coord_dec', **kwargs)
 
     def __call__(self, catalog, **kwargs):
-        return super(DecColumn, self).__call__(catalog, **kwargs)
+        return super().__call__(catalog, **kwargs)
 
 
 def fluxName(col):
```

Comment:
This function doesn't work anymore. If I do
 ```gPsFlux:
         functor: NanoMaggie
         args: slot_PsfFlux
         filt: HSC-G
         dataset: forced_src``` I get "[('forced_src', 'HSC-G', 'slot_PsfFlux_flux')] not in index"  and if I do: 
``` gPsFlux:
        functor: NanoMaggie
        args: slot_PsfFlux_instFlux
        filt: HSC-G
        dataset: forced_src``` I get 

"[('forced_src', 'HSC-G', 'slot_PsfFlux_instFlux_flux')] not in index"

Reply:
aahh, yes.  I should be testing this function.